src/Storages/MergeTree/ReplicatedMergeTreeTableMetadata.cpp

```diff
@@ -155,8 +159,18 @@ void ReplicatedMergeTreeTableMetadata::read(ReadBuffer & in)
     else
         index_granularity_bytes = 0;
 
-    if (checkString("constraints: ", in))
-        in >> constraints >> "\n";
+    String verb;
+    readStringUntilWhitespace(verb, in);
+
+    if (verb == "constraints:")
+    {
+        in >> " " >> constraints >> "\n";
+
+        readStringUntilWhitespace(verb, in);
```

Comment:
Isn't it possible for "constraints" to contain a whitespace?
I suggest to add test cases with constraints.